aws-s3-transfer-manager/tests/upload_test.rs

```diff
@@ -171,3 +178,35 @@ async fn test_many_uploads_no_deadlock() {
         handle.join().await.unwrap();
     }
 }
+
+#[tokio::test]
+async fn test_large_upload_part_size_bump() {
+    let (_guard, logs_rx) = capture_test_logs();
+    let client = mock_s3_client_for_multipart_upload();
+    let config = aws_s3_transfer_manager::Config::builder()
+        .client(client)
+        .build();
+
+    let tm = aws_s3_transfer_manager::Client::new(config);
+
+    let (tx, rx) = mpsc::channel(1);
+    let size_hint = 100 * ByteUnit::Gibibyte.as_bytes_u64();
+    let stream = TestStream::new(rx, 0, size_hint);
+
+    let handle = tm
+        .upload()
+        .bucket("test-bucket")
+        .key("test-large-upload-part-size".to_string())
+        .body(InputStream::from_part_stream(stream))
+        .initiate()
+        .unwrap();
+
+    // actual object is empty but we should have bumped the part_size from size_hint
+    drop(tx);
+    handle.join().await.unwrap();
+
+    let expected_part_size = 10737419;
```

Comment:
instead of the hard coded number, maybe use `size_hint.div_ceil(MAX_PARTS)`

Reply:
but the hard coded one doesn't make any sense. just it will be easy to read.

Checking against the expected logic is what we want for test

Reply:
> but the hard coded one doesn't make any sense, it looks like a weird number comes from nowhere.
I have added a comment

>  And checking against the expected logic is what we want for test

I think that we should test the expected logic with the expected value, to validate that the logic is correct. Copying the logic won't validate the logic because it will always be equal.

Reply:
However, I think this case is simple enough that we can go either way.